clients/admin-ui/src/features/privacy-requests/configuration/StorageConfiguration.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Box,
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  Heading,
+  Radio,
+  RadioGroup,
+  Stack,
+  Text,
+} from "@fidesui/react";
+import NextLink from "next/link";
+import { useState } from "react";
+
+import { getErrorMessage } from "~/features/common/helpers";
+import { useAlert } from "~/features/common/hooks";
+import Layout from "~/features/common/Layout";
+import {
+  useGetStorageDetailsQuery,
+  useSetActiveStorageMutation,
+} from "~/features/privacy-requests/privacy-requests.slice";
+
+import S3StorageConfiguration from "./S3StorageConfiguration";
+
+interface StorageData {
+  type: string;
+  details: {
+    auth_method: string;
+    bucket: string;
+  };
+  format: string;
+}
+
+const StorageConfiguration = () => {
+  const [existingStorageData, setExistingStorageData] = useState<StorageData>();
+  const { data: existingData } = useGetStorageDetailsQuery(
+    existingStorageData?.type
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const { errorAlert, successAlert } = useAlert();
+  const [saveStorageType] = useSetActiveStorageMutation();
+
+  const handleChange = async (value: string) => {
+    setIsLoading(true);
```

Comment:
One thing RTK gives us for free is an `isLoading` for all queries/mutations. For example: https://github.com/ethyca/fides/blob/b6808303639ac718804571cc6c97e1515e76b7c2/clients/admin-ui/src/features/config-wizard/AuthenticateAwsForm.tsx#L97